GOAT-Compiler/Code Generation/BuildInFunctionImplementations.cs

```diff
@@ -0,0 +1,343 @@
+        {
+            switch (functionName)
+            {
+                case "RelMove":
+                    RelMove(actuelParams[0]);
+                    break;
+                case "AbsMove":
+                    AbsMove(actuelParams[0]);
+                    break;
+                case "RelArcCW":
+                    RelArc(actuelParams[0], actuelParams[1], false);
+                    break;
+                case "RelArcCCW":
+                    RelArc(actuelParams[0], actuelParams[1], true);
+                    break;
+                case "AbsArcCW":
+                    AbsArc(actuelParams[0], actuelParams[1], false);
+                    break;
+                case "AbsArcCCW":
+                    AbsArc(actuelParams[0], actuelParams[1], true);
+                    break;
+                case "SetExtrusionRate":
+                    SetExtrusionRate(actuelParams[0]);
+                    break;
+                case "SetBedTemp":
+                    SetBedTemp(actuelParams[0]);
+                    break;
+                case "SetExtruderTemp":
+                    SetExtruderTemp(actuelParams[0]);
+                    break;
+                case "SetFanPower":
+                    SetFanPower(actuelParams[0]);
+                    break;
+                case "Steps":
+                    Steps(actuelParams[0]);
+                    break;
+                case "Position":
+                    return Position();
+                case "Lift":
+                    Lift(actuelParams[0]);
+                    break;
+                case "Right":
+                    Right(actuelParams[0]);
+                    break;
+                case "Left":
+                    Left(actuelParams[0]);
+                    break;
+                case "Direction":
+                    return Direction();
+                case "TurnTo":
+                    TurnTo(actuelParams[0]);
+                    break;
+                case "WaitForBedTemp":
+                    WaitForBedTemp();
+                    break;
+                case "WaitForExtruderTemp":
+                    WaitForExtruderTemp();
+                    break;
+                case "WaitForCurrentMove":
+                    WaitForCurrentMove();
+                    break;
+                case "WaitForMillis":
+                    WaitForMillis(actuelParams[0]);
+                    break;
+                case "Home":
+                    Home();
+                    break;
+            }
+            return null;
+        }
+
+        private void Home()
+        {
+            string gLine = "G28";
+            _machine.Position = new Vector(0, 0, 0);
+            _stream.WriteLine(gLine);
+        }
+
+        private void RelMove(Vector v)
+        {
+            string gLine = "";
+            Vector oldPosition = _machine.Position;
+            _machine.Position = new Vector(oldPosition.X + v.X, oldPosition.Y + v.Y, oldPosition.Z + v.Z);
```

Comment:
why not just _machine.Position +=v?

Reply:
Added operator overload on the vector class, "+" and "-"